### PR TITLE
Add support for signing with Tink keyset

### PR DESCRIPTION
**Pull request by Hayden-IO**
This allows deployers to securely sign in-memory while mitigating key exfiltration, since the key is encrypted at rest and loaded into memory at server startup.

Requires providing a path to an encrypted Tink keyset and the Key Encryption Key, a KMS URI for decrypting the keyset.

Heavily pulls from Fulcio's Tink implementation.

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
thanks
lgtm

File: cmd/rekor-server/app/root.go [hunk 1]
```diff
@@ -98,9 +98,11 @@ func init() {
 	rootCmd.PersistentFlags().String("rekor_server.address", "127.0.0.1", "Address to bind to")
 
 	rootCmd.PersistentFlags().String("rekor_server.signer", "memory",
-		`Rekor signer to use. Valid options are: [awskms://keyname, azurekms://keyname, gcpkms://keyname, hashivault://keyname, memory, <filename containing PEM-encoded private key>].
+		`Rekor signer to use. Valid options are: [awskms://keyname, azurekms://keyname, gcpkms://keyname, hashivault://keyname, memory, tink, <filename containing PEM-encoded private key>].
 Memory and file-based signers should only be used for testing.`)
 	rootCmd.PersistentFlags().String("rekor_server.signer-passwd", "", "Password to decrypt signer private key")
+	rootCmd.PersistentFlags().String("rekor_server.tink_kek_uri", "", "Key encryption key for decrypting Tink keyset. Valid options are [aws-kms://keyname, gcp-kms://keyname]")
```
Comment (bobcallaway) [inline cmd/rekor-server/app/root.go:104]: could we use `awskms` and `gcpkms` to match above?

Comment (Hayden-IO) [reply cmd/rekor-server/app/root.go:104]: No unfortunately, these are the values hardcoded in Tink.

Comment (Hayden-IO) [reply cmd/rekor-server/app/root.go:104]: https://github.com/tink-crypto/tink-go-gcpkms/blob/main/integration/gcpkms/gcp_kms_client.go#L33C2-L33C11

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**